app/controllers/notifications_controller.rb

```diff
@@ -66,4 +67,12 @@ def if_not_signed_in
         end
       end
     end
+
+    # Stops here if the signed in user is not the owner of
+    # the notification set by :set_notification.
+    def ensure_user_owns_notification!
+      unless @notification.user == current_user
```

Comment:
I'm also pretty new to open source and I would like to give some cents too LOL.

1 - Since both callbacks are: `only: [:destroy]` does it make sense to have both of them? I think retrieving the record + checking the user in the destroy action is not that bad. 
2 - I know it is not related to this pull request but this endpoint should be only for logged users. We could get rid of the `def signed_in` method and put the current_user in a base controller.

Then the action destroy could be something like:

`current_user.notifications.where(id: params[:id]) `
or
`Notifications.where(id: params[:id], user: current_user)`

IMHO -> For now I would do the whole thing in the action (probably using some private helper methods), but without callbacks.